core/src/main/scala/org/apache/spark/sql/TiStrategy.scala

```diff
@@ -115,7 +130,16 @@ case class TiStrategy(getOrCreateTiContext: SparkSession => TiContext)(sparkSess
 
   override def apply(plan: LogicalPlan): Seq[SparkPlan] = {
     val ts = tiContext.tiSession.getTimestamp
-    plan foreachUp applyStartTs(ts)
+
+    if (plan.isStreaming) {
```

Comment:
I feel like we need some comment to explain why we do this.

Reply:
nice!